CONTRIB-6213 mod_surveypro: make code simpler

kordan opened the pull request:
During  code cleanup I found that
- surveypro_fetch_items_seeds was more complex than needed
- item_validate_variablename was more complex than needed too
and
- get_used_plugin_list was written three times in the code.

item_validate_variablename had also an error (with very low probability to occur, but my Editing Teacher are really nice :o) )

Yes, they should be three PR but they are all small changes.
I apologise in advance.

kordan commented:
behat tests were all passing

stronk7 commented:
Just if you fix the commit message... (missing "C").

kordan commented:
Not sure about the missing starting "C"
:-P
;-)

kordan commented on classes/itembase.class.php:
```diff
@@ -480,42 +480,40 @@ public function item_validate_variablename($record, $itemid) {
-            $candidatevariable = $record->variable;
+            $testname = $record->variable;
         }
+        // Bloody Editing Teachers were creating a boolean element naming it 'age_001'
```
Did you see what a crazy error my ET found?

kordan commented:
Bloody hell!

git co simplify_code
git rebase -i master
i
reword
esc :wq
i
Add the C
esc :wq
git push origin simplify_code --force

I am sure about this but here, in github, I can not see the "C"

My git log --oneline is
MacBookProKordan:surveypro kordan$ git log --oneline 
19ade24 CONTRIB-6213 mod_surveypro: make code simpler
5adbb49 Merge pull request #58 from kordan/code_cleanup_2016032401
c249578 NOBUG mod_surveypro: code cleanup

WHY? GRRRR

kordan commented:
GRRRRR

MacBookProKordan:surveypro kordan$ git rebase -i master
[detached HEAD d8f8de7] CCCCONTRIB-6213 mod_surveypro: make code simpler
 Author: Kordan <kordan@mclink.it>
 17 files changed, 126 insertions(+), 104 deletions(-)
Successfully rebased and updated refs/heads/simplify_code.
MacBookProKordan:surveypro kordan$ git push origin simplify_code --force
fatal: cannot exec 'git-credential-osxkeychain': Bad CPU type in executable
Username for 'https://github.com': kordan
Password for 'https://kordan@github.com': 
fatal: cannot exec 'git-credential-osxkeychain': Bad CPU type in executable
Counting objects: 1, done.
Writing objects: 100% (1/1), 461 bytes | 0 bytes/s, done.
Total 1 (delta 0), reused 0 (delta 0)
To https://github.com/kordan/moodle-mod_surveypro.git
 + 19ade24...d8f8de7 simplify_code -> simplify_code (forced update)

It seems to be fine but then the C is still missing!!!!

kordan commented:
I give up! I changed it using the button "Edit" of this gitbub web page at the top rigth corner.
:(

stronk7 commented:
But... Daniele... all I was asking was you to change the commit message from "ONTRIB..." to "CONTRIB", lol. ( commit --amend + push --force).Not sure what's current "CCCONTRIB". Sorry cannot stop joking.

kordan commented:
I did it once again but I have no evidence it really did what it was supposed to do.
How can I verify the execution was fine?

stronk7 commented:
Sorry Daniele, but... what are you trying to verify? I'm lost.

Your last commit sent here seems to be ok. It has the correct "CONTRIB" in the commit message. Previously it had incorrect "ONTRIB" and later "CCCONTRIB", and that is what I was asking about.

So, what am I missing, which "execution" are you talking?

Ciao :-)

kordan commented:
Before sending out a work, I usually test it, wearing for a moment the dresses of the person I am talking with.
Last week, I issued git rebase -i master bla bla bla and then I arived here (https://github.com/kordan/moodle-mod_surveypro/pull/59) to verify that the result was actually reached. What I found was that in the title of this page (where now it is written "CONTRIB-6213 mod_surveypro: make code simpler #59") I still had "ONTRIB-6213 mod_surveypro: make code simpler #59". This is why I rebased again and I added four times the starting C. Because I was angry!
Yesterday I issued git commit --amend and I changed CCCC to C. My question was: "where can I go to check if the correction really fixed the comment?"

kordan commented:
But before all: Good Easter Monday!

stronk7 commented:
I think github works this way:

1) When the pull request is **created**, the commit message is used for the title. That's the cause the title of this issue was "ONTRIB", because your commit message was, also, "ONTRIB".
2) After creation, both commit messages and PR titles become **unconnected**. You can change the commit message and that won't change the PR title (you did that). And you can edit the PR and the commit won't change. They are 2 completely separated entities.

So, I think that you were expecting the title to be changed automatically and that does not happen. Just the commits do. As you can see above the commit associated with this PR is the correct one (commits tab). And also the title, but because you edited it manually.

Heh, same happy Monday there!

kordan commented:
Yes, I think your interpretation is 100% correct.
I expected a feedback from this web page that was not supposed to provide it.
In spite of this, what is nice is that I had only one way to change the commit message (rebase -i) while now I have two ways to change it. I used it but I never focused my attention on it. :)

stronk7 commented:
merged, thanks!

kordan commented:
\o/
Thaaaaanks
